src/txn/txn_rollback_to_stable.c

```diff
@@ -1468,33 +1471,41 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
             continue;
         }
 
-        /* Set this flag to return error instead of panic if file is corrupted. */
-        F_SET(session, WT_SESSION_QUIET_CORRUPT_FILE);
-        ret = __wt_session_get_dhandle(session, uri, NULL, NULL, 0);
-        F_CLR(session, WT_SESSION_QUIET_CORRUPT_FILE);
-
-        /*
-         * Ignore performing rollback to stable on files that does not exist or the files where
-         * corruption is detected.
-         */
-        if ((ret == ENOENT) ||
-          (ret == WT_ERROR && F_ISSET(S2C(session), WT_CONN_DATA_CORRUPTION))) {
-            __wt_verbose(session, WT_VERB_RECOVERY_RTS(session),
-              "ignore performing rollback to stable on %s because the file %s", uri,
-              ret == ENOENT ? "does not exist" : "is corrupted.");
-            continue;
-        }
-        WT_ERR(ret);
-
         /*
          * The rollback operation should be performed on this file based on the following:
-         * 1. The tree is modified.
+         * 1. The dhandle is present in the cache and tree is modified.
          * 2. The checkpoint durable start/stop timestamp is greater than the rollback timestamp.
          * 3. There is no durable timestamp in any checkpoint.
          * 4. The checkpoint newest txn is greater than snapshot min txn id
          */
-        if (S2BT(session)->modified || max_durable_ts > rollback_timestamp || prepared_updates ||
-          !durable_ts_found || has_txn_updates_gt_than_ckpt_snap) {
+        WT_WITH_HANDLE_LIST_READ_LOCK(session, (ret = __wt_conn_dhandle_find(session, uri, NULL)));
```

Comment:
Accessing the connection dhandle function breaks any function modularity? @bvpvamsikrishna, @agorrod

Reply:
It is uncommon to access the dhandle from the connection list directly, but in this case the intention is to bypass session cache, because RTS is not session specific. I will say it is okay to break convention here, and it doesn't break the modularity.